src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java

```diff
@@ -98,33 +99,57 @@ private CallbackResponse handleDefendAllClaim(CallbackParams callbackParams) {
-                && caseData.getRespondToClaim().getHowMuchWasPaid() != null
-                && caseData.getRespondToClaim().getHowMuchWasPaid().compareTo(caseData.getTotalClaimAmount()) >= 0) {
-                caseData = caseData.toBuilder()
-                    .respondent1ClaimResponsePaymentAdmissionForSpec(
-                        RespondentResponseTypeSpecPaidStatus.PAID_FULL_OR_MORE_THAN_CLAIMED_AMOUNT).build();
-            }
-
-            AllocatedTrack allocatedTrack = AllocatedTrack.getAllocatedTrack(caseData.getTotalClaimAmount(),
-                                                                             null);
+    //TODO Duplication Check for the methods to see if it can have a generic ways in the future
+    private CallbackResponse handleAdmitPartOfClaim(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        List<String> errors = paymentDateValidator.validate(Optional.ofNullable(caseData.getRespondToAdmittedClaim())
+                                                                .orElseGet(() -> RespondToClaim.builder().build()));
+        if (!errors.isEmpty()) {
             return AboutToStartOrSubmitCallbackResponse.builder()
-                .data(caseData.toBuilder().responseClaimTrack(allocatedTrack.name()).build().toMap(objectMapper))
+                .errors(errors)
                 .build();
         }
+        if (SpecJourneyConstantLRSpec.DEFENDANT_RESPONSE_SPEC.equals(callbackParams.getRequest().getEventId())) {
+            return populateTotalClaimAmount(caseData);
+        }
         return AboutToStartOrSubmitCallbackResponse.builder()
             .data(caseData.toBuilder().build().toMap(objectMapper))
             .build();
     }
 
+    private void populateRespondentResponseTypeSpecPaidStatus(CaseData caseData) {
+        if (SpecJourneyConstantLRSpec.HAS_PAID_THE_AMOUNT_CLAIMED.equals(caseData.getDefenceRouteRequired())
+            && caseData.getRespondToClaim().getHowMuchWasPaid() != null
+            && caseData.getRespondToClaim().getHowMuchWasPaid().compareTo(caseData.getTotalClaimAmount()) < 0) {
+            caseData = caseData.toBuilder()
+                .respondent1ClaimResponsePaymentAdmissionForSpec(
+                    RespondentResponseTypeSpecPaidStatus.PAID_LESS_THAN_CLAIMED_AMOUNT).build();
+        } else if (SpecJourneyConstantLRSpec.HAS_PAID_THE_AMOUNT_CLAIMED
+            .equals(caseData.getDefenceRouteRequired())
+            && caseData.getRespondToClaim().getHowMuchWasPaid() != null
+            && caseData.getRespondToClaim().getHowMuchWasPaid().compareTo(caseData.getTotalClaimAmount()) >= 0) {
+            caseData = caseData.toBuilder()
+                .respondent1ClaimResponsePaymentAdmissionForSpec(
+                    RespondentResponseTypeSpecPaidStatus.PAID_FULL_OR_MORE_THAN_CLAIMED_AMOUNT).build();
+        }
+
+    }
+
+    private CallbackResponse populateTotalClaimAmount(CaseData caseData) {
```

Comment:
This does not populate claim amount. rather it populates Allocated track - Fast track or Small claim. Please name the method appropriately with something like populateAllocatedTrack